src/components/DataProvider/index.js

```diff
@@ -157,6 +154,32 @@ export default class DataProvider extends Component {
     if (reason !== 'UPDATE_SUBDOMAIN') {
       update.contextSeries = series;
     }
+    if (reason === 'INTERVAL') {
+      let domainMax = this.state.domain[1];
+      Object.keys(series).forEach(key => {
+        const { data, xAccessor } = series[key];
+        const serieXMax = xAccessor(data[data.length - 1]);
+        if (serieXMax > domainMax) {
+          domainMax = serieXMax;
+        }
+      });
+
+      update.domain = [this.state.domain[0], domainMax];
+
+      // follow most right subdomain
+      if (this.state.subDomain[1] === this.state.domain[1]) {
+        update.subDomain = [this.state.subDomain[0], domainMax];
+      }
+
+      // shift serie data array if a new item has been added to the end
+      if (this.props.shift) {
+        Object.keys(series).forEach(key => {
+          if (series[key].data.length > this.state.series[key].data.length) {
+            series[key].data.shift();
```

Comment:
Don't we need to to call `shift()` the number of times as however many elements were added?

Reply:
Good catch